code-challenges.js

```diff
@@ -30,8 +79,30 @@ var randomNouns = ["streetlamp", "potato", "teeth", "conclusion", "nephew", "tem
 var mixedDataArray = [true, 8, "hello", 90, -8, null, 0, 46, 59, 107, "hey!"]
 // Expected output: [-8, 0, 8, 46, 59, 90, 107]
 
-
-
+//thoughts:
+//at first glance it seems like we'd have to use filter
+//we used the sort filter yesterday during office hours in pig latin, I'm gonna experiment with it for this problem.
+    //sort with no arguments arranges it not by the number as a whole, but by the first number so it would consider 100 less than 4 (b/c 1 < 4).
+
+//create a function that takes in an array 
+const numsOnly = (array) => {
+    //use filter method to take in the values of the array and return a new array that contains only numbers
+    let numsArr = array.filter(value => {
+        //use typeof === "number" to only return values that are numbers
+        return (typeof value === "number")
+    })    
+    //use the sort method to sort the numbers from least to greatest.
+    return numsArr.sort(function(a, b){return a-b});
+    //research on sort()
+        //so, to make the sort method look at the entire number and not just the first digit we need to send it into a "compareFunction" that lives inside the sort() method, it sends two values from the array and sorts it according to the returned value (positive, negative, or zero)
+        //to visualize when nums Arr is first created it is [ 8, 90, -8, 0, 46, 59, 107 ]
+        //two values are taken (a = 8) (b = 90)
+        //we return the value of a - b (8 - 90 = -82), since it is negative a < b, and moves the value of a to the left. (this function basically moves the smaller number to the left until it is the larger number in the compareFunction)
+            //if I entered {return b-a} since (90 - 8 = 82) it's positive and will move 8 to the right etc etc. until it gets sorted from greatest to least
```

Comment:
Yes! Great job explaining out .sort and how it works. You've got the idea down well!